librz/core/cmd_descs/cmd_debug.yaml

```
@@ -312,78 +324,194 @@ commands:
-            summary: Step until <addr>
-            args:
-              - name: addr
-                type: RZ_CMD_ARG_TYPE_RZNUM
-          - name: dsui
-            cname: cmd_debug_step_until_instr
-            summary: Step until an instruction that matches <instr>
-            args:
-              - name: instr
-                type: RZ_CMD_ARG_TYPE_STRING
-          - name: dsuir
-            cname: cmd_debug_step_until_instr_regex
-            summary: Step until an instruction that matches <regex>
-            args:
-              - name: regex
-                type: RZ_CMD_ARG_TYPE_STRING
-          - name: dsuo
-            cname: cmd_debug_step_until_optype
-            summary: Step until an instruction matches one of the <optype>s
-            args:
-              - name: optype
-                type: RZ_CMD_ARG_TYPE_STRING
-                flags: RZ_CMD_ARG_FLAG_ARRAY
-          - name: dsue
-            cname: cmd_debug_step_until_esil
-            summary: Step until <esil> expression matches
-            args:
-              - name: esil
-                type: RZ_CMD_ARG_TYPE_STRING
-          - name: dsuf
-            cname: cmd_debug_step_until_flag
-            summary: Step until pc == <flag> matching name
-            args:
-              - name: flag
-                type: RZ_CMD_ARG_TYPE_FLAG
-  - name: dts
-    summary: Debug trace session commands
+      - name: dd
+        summary: >
+          List file descriptors or Open
+          / Open and map that file into the UI
+        cname: cmd_debug_dd
+        args:
+          - name: file
+            type: RZ_CMD_ARG_TYPE_FILE
+            optional: true
+      - name: dd-
+        summary: Close stdout fd
+        cname: cmd_debug_fd_close
+        args:
+          - name: fd
+            type: RZ_CMD_ARG_TYPE_NUM
+      - name: dd*
+        summary: List file descriptors (in rizin commands)
+        cname: cmd_debug_list_fd
+        args: []
+      - name: dds
+        summary: Seek given fd
+        cname: cmd_debug_fd_seek
+        args:
+          - name: fd
+            type: RZ_CMD_ARG_TYPE_NUM
+          - name: offset
+            type: RZ_CMD_ARG_TYPE_NUM
+      - name: ddd
+        summary: Dup2 from fd1 to fd2
+        cname: cmd_debug_dup2
+        args:
+          - name: fd1
+            type: RZ_CMD_ARG_TYPE_NUM
+          - name: fd2
+            type: RZ_CMD_ARG_TYPE_NUM
+      - name: ddr
+        summary: Read N bytes from fd
+        cname: cmd_debug_fd_read
+        args:
+          - name: fd
+            type: RZ_CMD_ARG_TYPE_NUM
+          - name: offset
+            type: RZ_CMD_ARG_TYPE_NUM
+          - name: size
+            type: RZ_CMD_ARG_TYPE_NUM
+      - name: ddw
+        summary: Write N bytes to fd
+        cname: cmd_debug_fd_write
+        args:
+          - name: fd
+            type: RZ_CMD_ARG_TYPE_NUM
+          - name: offset
+            type: RZ_CMD_ARG_TYPE_NUM
+          - name: len
+            type: RZ_CMD_ARG_TYPE_NUM
```

```suggestion
          - name: offset
            type: RZ_CMD_ARG_TYPE_RZNUM
          - name: size
            type: RZ_CMD_ARG_TYPE_RZNUM
```
use rz_num_math in the handler.